fbw-common/src/wasm/systems/systems/src/hydraulic/flap_slat.rs

```diff
@@ -666,472 +758,542 @@ mod tests {
+        assert!(test_bed.right_motor_speed() == AngularVelocity::ZERO);
+        assert!(test_bed.left_motor_flow() == VolumeRate::ZERO);
+        assert!(test_bed.right_motor_flow() == VolumeRate::ZERO);
+        assert!(test_bed.synchro_position() == Angle::ZERO);
+    }
+
+    #[test]
+    fn flap_slat_assembly_variables_check() {
+        let mut test_bed = test_bed();
+
+        let flap_position_request = Angle::new::<degree>(20.);
+        test_bed = test_bed
+            .set_angle_request(Some(flap_position_request))
+            .set_hyd_pressure(Pressure::new::<psi>(MAX_CIRCUIT_PRESSURE_PSI));
+
+        test_bed.run_with_delta(Duration::from_millis(20000));
+
+        test_bed.check_flaps_position(flap_position_request);
+
+        let flap_position_request = Angle::new::<degree>(40.);
+        test_bed = test_bed.set_angle_request(Some(flap_position_request));
+
+        test_bed.run_with_delta(Duration::from_millis(20000));
+
+        test_bed.check_flaps_position(flap_position_request);
     }
 
     #[test]
     fn flap_slat_assembly_full_pressure() {
-        let max_speed = AngularVelocity::new::<radian_per_second>(0.11);
-        let mut test_bed = SimulationTestBed::new(|context| TestAircraft::new(context, max_speed));
+        let mut test_bed = test_bed();
 
-        assert!(test_bed.query(|a| a.flaps_slats.position_feedback().get::<degree>()) == 0.);
-        assert!(test_bed.query(|a| a.flaps_slats.speed.get::<radian_per_second>()) == 0.);
+        assert!(test_bed.synchro_position() == Angle::ZERO);
+        assert!(test_bed.flap_slat_speed() == AngularVelocity::ZERO);
 
-        test_bed.command(|a| a.set_angle_request(Some(Angle::new::<degree>(20.))));
-        test_bed.command(|a| {
-            a.set_current_pressure(
-                Pressure::new::<psi>(MAX_CIRCUIT_PRESSURE_PSI),
-                Pressure::new::<psi>(MAX_CIRCUIT_PRESSURE_PSI),
-            )
-        });
+        let flap_position_request = Angle::new::<degree>(20.);
+        test_bed = test_bed
+            .set_angle_request(Some(flap_position_request))
+            .set_hyd_pressure(Pressure::new::<psi>(MAX_CIRCUIT_PRESSURE_PSI));
 
         test_bed.run_with_delta(Duration::from_millis(2000));
 
-        let current_speed = test_bed.query(|a| a.flaps_slats.speed);
+        let current_speed = test_bed.flap_slat_speed();
         assert!(
-            (current_speed - max_speed).abs() <= AngularVelocity::new::<radian_per_second>(0.01)
+            (current_speed - test_bed.get_max_speed()).abs()
+                <= AngularVelocity::new::<radian_per_second>(0.01)
         );
 
         assert!(
-            test_bed.query(|a| a.flaps_slats.left_motor.speed())
-                >= AngularVelocity::new::<revolution_per_minute>(2000.)
-                && test_bed.query(|a| a.flaps_slats.left_motor.speed())
+            test_bed.left_motor_speed() >= AngularVelocity::new::<revolution_per_minute>(2000.)
+                && test_bed.left_motor_speed()
```

Comment:
Can we split this up into 2 asserts? This way in case of failure it's easier to see which condition failed.
(Same for the others below)